### ISSUE TITLE
Query params cannot be shared by nested routes

### ISSUE THREAD

**Issue by mdentremont** [emberjs/ember.js#13051]
I am trying to share query parameters between a parent and child route. Both routes use the same controller, which has query params defined on it.

I am seeing the following issues:
- The query params are cleared when trying to hit the nested route
- The query params are not set on the controller until the model is finished loading (I filed an issue for this here: https://github.com/emberjs/ember.js/issues/13052)

I created an ember-twiddle here to demonstrate: https://ember-twiddle.com/4ee7cc8a3446cbda4e11?route=%2Findex%2Fnested

**Comment by pixelhandler** [comment]
@mdentremont I cloned your ember-twiddle and added console.log in the model hooks of both the parent and nested routes. https://ember-twiddle.com/abd90f120bd434e3cfeb?openFiles=index.nested.route.js%2C&route=%2Findex%2Fnested%3Ftest%3DI%2520WILL%2520BE%2520CLEARED

see the console, both model hooks to get the same param key/value. the controller isn't setup until the model hooks complete so the delay results in rendering the default value you setup on the controller. 

I commented on #13052 regarding the default behavior of query params, route and controller lifecycle in Ember.
